pypdf/annotations.py

```diff
@@ -25,54 +37,45 @@ def _get_bounding_rectangle(vertices: List[Tuple[float, float]]) -> RectangleObj
     return rect
 
 
-class AnnotationBuilder:
-    """
-    The AnnotationBuilder creates dictionaries representing PDF annotations.
+class AnnotationDictionary(DictionaryObject, ABC):
+    def __init__(self) -> None:
+        # "rect" should not be added here as PolyLine can automatically set it
+        self[NameObject("/Type")] = NameObject("/Annot")
 
-    Those dictionaries can be modified before they are added to a PdfWriter
-    instance via ``writer.add_annotation``.
 
-    See `adding PDF annotations <../user/adding-pdf-annotations.html>`_ for
-    it's usage combined with PdfWriter.
+class Text(AnnotationDictionary):
+    """
+    A text annotation.
+
+    Args:
+        rect: array of four integers ``[xLL, yLL, xUR, yUR]``
+            specifying the clickable rectangular area
+        text: The text that is added to the document
+        open:
+        flags:
     """
 
-    from ..types import FitType, ZoomArgType
-
-    @staticmethod
-    def text(
+    def __init__(
+        self,
+        *,
```

Comment:
I would have personnally leave rect and text as positional parameters : many users may like this